lib/matplotlib/axes/_axes.py

```diff
@@ -7369,12 +7396,44 @@ def _kde_method(X, coords):
             kde = mlab.GaussianKDE(X, bw_method)
             return kde.evaluate(coords)
 
-        vpstats = cbook.violin_stats(dataset, _kde_method, points=points)
-        return self.violin(vpstats, positions=positions, vert=vert,
+        def _resolve_duplicate_alias(func_obj_list):
+            unique_alias_set = set()
+            for func_obj in func_obj_list:
+                while func_obj.alias in unique_alias_set:
+                    func_obj.alias += 'x'
+                unique_alias_set.add(func_obj.alias)
+
+        violin_stat_func_obj_list = []
+        for func in statistics_function_list:
+            if not isinstance(func, ViolinStatFunc):
```

Comment:
if `_ViolinStatFunc` is private, then we can only deal with callables and simplify this logic.